src/transformers/onnx/__main__.py

```diff
@@ -11,60 +11,61 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import subprocess
+import sys
 from argparse import ArgumentParser
 from pathlib import Path
 
-from ..models.auto import AutoFeatureExtractor, AutoProcessor, AutoTokenizer
-from ..onnx.utils import get_preprocessor
-from ..utils import logging
+from packaging import version
+
+from .. import AutoFeatureExtractor, AutoTokenizer, AutoProcessor
 from .convert import export, validate_model_outputs
+from ..utils import logging
+from ..utils.import_utils import is_optimum_available
+from .utils import get_preprocessor
 from .features import FeaturesManager
 
 
+MIN_OPTIMUM_VERSION = "1.5.0"
+
 ENCODER_DECODER_MODELS = ["vision-encoder-decoder"]
 
 
-def main():
-    parser = ArgumentParser("Hugging Face Transformers ONNX exporter")
-    parser.add_argument(
-        "-m", "--model", type=str, required=True, help="Model ID on huggingface.co or path on disk to load model from."
-    )
-    parser.add_argument(
-        "--feature",
-        choices=list(FeaturesManager.AVAILABLE_FEATURES),
-        default="default",
-        help="The type of features to export the model with.",
-    )
-    parser.add_argument("--opset", type=int, default=None, help="ONNX opset version to export the model with.")
-    parser.add_argument(
-        "--atol", type=float, default=None, help="Absolute difference tolerance when validating the model."
-    )
-    parser.add_argument(
-        "--framework",
-        type=str,
-        choices=["pt", "tf"],
-        default=None,
-        help=(
-            "The framework to use for the ONNX export."
-            " If not provided, will attempt to use the local checkpoint's original framework"
-            " or what is available in the environment."
-        ),
-    )
-    parser.add_argument("output", type=Path, help="Path indicating where to store generated ONNX model.")
-    parser.add_argument("--cache_dir", type=str, default=None, help="Path indicating where to store cache.")
-    parser.add_argument(
-        "--preprocessor",
-        type=str,
-        choices=["auto", "tokenizer", "feature_extractor", "processor"],
-        default="auto",
-        help="Which type of preprocessor to use. 'auto' tries to automatically detect it.",
+def export_with_optimum(args):
+    if is_optimum_available():
+        from optimum.version import __version__ as optimum_version
+
+        parsed_optimum_version = version.parse(optimum_version)
+        if parsed_optimum_version < version.parse(MIN_OPTIMUM_VERSION):
+            raise RuntimeError(
+                f"transformers.onnx requires optimum >= {MIN_OPTIMUM_VERSION} but {optimum_version} is installed. You "
+                "can upgrade optimum by running: pip install -U optimum"
+            )
+    else:
+        raise RuntimeError(
+            "transformers.onnx requires optimum to run, you can install the library by running: pip install optimum"
```

Comment:
Same suggestion as above:
```suggestion
            "transformers.onnx requires optimum to run, you can install the library by running: pip install optimum[onnxruntime]"
```